arithmetization/src/main/java/net/consensys/linea/zktracer/opcode/OpCodes.java

```diff
@@ -32,27 +33,29 @@
 public class OpCodes {
   private static final JsonConverter YAML_CONVERTER = JsonConverter.builder().enableYaml().build();
 
-  private static Map<Integer, OpCodeData> valueToOpCodeDataMap;
-  private static Map<OpCode, OpCodeData> opCodeToOpCodeDataMap;
+  private static Map<Integer, OpCodeData> valueToOpCodeDataMap = new ConcurrentHashMap<>();
+  private static Map<OpCode, OpCodeData> opCodeToOpCodeDataMap = new ConcurrentHashMap<>();
 
   /** Loads all opcode metadata from src/main/resources/opcodes.yml. */
   @SneakyThrows(IOException.class)
   public static void load() {
-    JsonNode rootNode =
-        YAML_CONVERTER
-            .getObjectMapper()
-            .readTree(OpCodes.class.getClassLoader().getResourceAsStream("opcodes.yml"))
-            .get("opcodes");
+    if (valueToOpCodeDataMap.isEmpty() || opCodeToOpCodeDataMap.isEmpty()) {
+      JsonNode rootNode =
+          YAML_CONVERTER
+              .getObjectMapper()
+              .readTree(OpCodes.class.getClassLoader().getResourceAsStream("opcodes.yml"))
```

Comment:
shoudn't we move the opcodes.yml and spillings.toml to outside the jar and treat them as a config file? @fab-10 , any opinion?

Reply:
idk about those file, better ask to the arithimization team, because if they do not need to be configured at runtime, it is fine to threat them as embedded resources